local_settings.sh.in

```diff
@@ -16,45 +16,98 @@ MODULES=${MODULES:-@MODULES@}
 # See bug for bug #1560019 for more information.
 declare -A file_modes
 
+# This associative array ("dict", "map", "hash") holds ALL the specific
+# location with specific fcontext settings.
+declare -A custom_fcontext=(
+["$SHAREDSTATEDIR/designate/bind9(/.*)?"]='named_zone_t'
+["$SHAREDSTATEDIR/vhost_sockets(/.*)?"]='virt_cache_t'
+["$SHAREDSTATEDIR/openstack-dashboard"]='httpd_var_lib_t'
+["$SHAREDSTATEDIR/mongodb(/.*)?"]='mongod_var_lib_t'
+["$LOCALSTATEDIR/log/gnocchi/app.log"]='httpd_log_t'
+["$LOCALSTATEDIR/log/aodh/app.log"]='httpd_log_t'
+["$LOCALSTATEDIR/log/ceilometer/app.log"]='httpd_log_t'
+["$LOCALSTATEDIR/log/panko/app.log"]='httpd_log_t'
+["$LOCALSTATEDIR/log/zaqar/zaqar.log"]='httpd_log_t'
+["$LOCALSTATEDIR/log/containers(/.*)?"]='container_file_t'
+["$LOCALSTATEDIR/lib/config-data(/.*)?"]='container_file_t'
+["$LOCALSTATEDIR/lib/tripleo-config(/.*)?"]='container_file_t'
+["$BINDIR/neutron-rootwrap-daemon"]='neutron_exec_t'
+["$BINDIR/neutron-vpn-agent"]='neutron_exec_t'
+["$LOCALSTATEDIR/cache/swift(/.*)"]='swift_var_cache_t'
+["$BINDIR/swift-object-reconstructor"]='swift_exec_t'
+["$BINDIR/swift-object-relinker"]='swift_exec_t'
+["${ROOTDIR}httpboot(/.*)?"]='httpd_sys_content_t'
+["$SHAREDSTATEDIR/nova/.ssh(/.*)?"]='ssh_home_t'
+["${ROOTDIR}tftpboot(/.*)?"]='tftpdir_t'
+["$LOCALSTATEDIR/log/pacemaker.log.*"]='cluster_var_log_t'
+["$LOCALSTATEDIR/log/pacemaker(/.*)?"]='cluster_var_log_t'
+)
+
+# This associative array ("dict", "map", "hash") will holds all the current
+# custom fcontext settings
+declare -A existing_custom_fcontext
+
+# This associative array ("dict", "map", "hash") will holds all the differences
+# we have between existing custom fcontext, and what we really want.
+# It will be used in different methods in this script.
+declare -A diff_fcontext
+
+# Let's poppulate the existing_fcontext
+for l in $(semanage fcontext -Cln | awk -F '[: ]+' '{print $1";"$6}'); do
+	existing_fcontext+=([$(echo $l|cut -d ';' -f1)]=$(echo $l|cut -d ';' -f2))
```

Comment:
Should this be existing_custom_fcontext too?

Reply:
not only here in fact - I probably messed up my git repo last time and pushed the wrong content..